Fix CUI `setcellwidths` characters draw behavior to same GUI behavior.

mikoto2000 opened the pull request:
Fixed #14539 issue.

mikoto2000 committed:
Fix CUI `setcellwidths` characters draw behavior to same GUI behavior.

mikoto2000 committed:
Add test When `setcellwidth` is used on characters that are not targets of `ambiwidth`.

zeertzjq commented on src/mbyte.c:
```diff
@@ -140,6 +140,7 @@ static int dbcs_head_off(char_u *base, char_u *p);
 #ifdef FEAT_EVAL
 static int cw_value(int c);
 #endif
+int get_cellwidth(int c);
```
This is not needed because it's a non-static function.

mikoto2000 replied:
Thank you.
Fixed in commit c69bff00c240028442255f77ee861421e5100db5.

mikoto2000 committed:
Delete unneeded prototype declarations.

mikoto2000 committed:
Fix unused parameter error.

Added `UNUSED` to the parameter `c` of `get_cellwidth` because it is not used when the build flag `FEAT_EVAL` is not defined.
